firefox-ios/Client/Frontend/Share/ShareManager.swift

```diff
@@ -76,104 +65,70 @@ class ShareManager: NSObject, FeatureFlaggable {
         return activityViewController
     }
 
-    /// Get the data to be shared if the URL is a file we will share just the url if not we prepare
-    /// UIPrintInfo to get the option to print the page and tab URL and title
-    /// - Parameter url: url from the selected tab
-    /// - Returns: An array of elements to be shared
-    private func getActivityItems(url: URL) -> [Any] {
-        // If url is file return only url to be shared
-        guard !url.isFileURL else { return [url] }
+    static func getActivityItems(
+        forShareType shareType: ShareType,
+        withExplicitShareMessage explicitShareMessage: ShareMessage?
+    ) -> [Any] {
+        var activityItems: [Any] = []
 
-        var activityItems = [Any]()
+        switch shareType {
+        case .file(let fileURL):
+            activityItems.append(URLActivityItemProvider(url: fileURL))
 
-        // Add the title (if it exists)
-        if let title = self.title {
-            activityItems.append(title)
-        }
+            if let explicitShareMessage {
+                activityItems.append(TitleSubtitleActivityItemProvider(shareMessage: explicitShareMessage))
+            }
+
+        case .site(let siteURL):
+            activityItems.append(URLActivityItemProvider(url: siteURL))
 
-        let printInfo = UIPrintInfo(dictionary: nil)
-        printInfo.jobName = (url.absoluteString as NSString).lastPathComponent
-        printInfo.outputType = .general
-        activityItems.append(printInfo)
-
-        // when tab is not loaded (webView != nil) don't show print activity
-        if let tab = selectedTab, tab.webView != nil {
-            activityItems.append(
-                TabPrintPageRenderer(
-                    tabDisplayTitle: tab.displayTitle,
-                    tabURL: tab.url,
-                    webView: tab.webView
+            // For websites shared from a place without a webview (e.g. bookmarks), we don't actually have webview to offer
+            // any advanced information (like title, printing, sent to the iOS home screen, etc.)
+            if let explicitShareMessage {
+                activityItems.append(TitleSubtitleActivityItemProvider(shareMessage: explicitShareMessage))
+            }
+
+        case .tab(let siteURL, let tab):
+            // For websites, we also want to offer a few additional activity items besides the URL, like printing the
+            // webpage or adding a website to the iOS home screen
+            activityItems.append(URLActivityItemProvider(url: siteURL))
+
+            // Only show the print activity if the tab's webview is loaded
+            if tab.webView != nil {
```

Comment:
nit; Could be a guard instead?

Reply:
I think this is ok here. Since it's not a "fatal" error to not have a loaded webView, a guard might be misleading for the intention. We want to continue with appending the below elements regardless of whether or not the webview is loaded.